errors/base-error.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const ExtendableError = require('es6-error')
+const isObject = require('101/is-object')
+
+/**
+ * Extendable base class that includes additional information that effects the
+ * way errors are reported to rollbar.
+ * @class
+ * @author Ryan Sandor Richards
+ * @extends es6-error~ExtendableError
+ */
+module.exports = class BaseError extends ExtendableError {
+  /**
+   * Creates a new base error.
+   * @param {String} message Message for the error.
+   * @param {Object} data Additional data for the error to be given to rollbar.
+   * @param {Object} reporting Reporting options.
+   */
+  constructor (message, data, reporting) {
+    super(message)
+    this.data = isObject(data) ? data : {}
+    this.reporting = isObject(reporting) ? reporting : {}
```

Comment:
Actually, this variable name is in all of the classes and it's very confusing what it actually does. I'd be fine if this would only be used internally, but it's a parameter I need to pass to create every single type of error. 

For example:

If I want to pass the queue or the job into a `WorkerError` I actually need to figure out what the `reporting` parameter is because I hate passing an ugly `undefined` into my functions. I know it's an object  because of JSDoc (I'm already upset I had to look at the JSDoc instead of having proper variable naming), but JSDoc doesn't actually document what the parameters of the object actually are. I have to go in to the `BaseWorker` to figure out that this only has two properties: `level` and `fingerprint`. Now I'm really upset because I don't even need those two properties and wonder why the hell there's a parameter for that in first place (isn't that why we're using classes in the first place).

My suggestion is to get rid of this parameter completely and just use the `setLevel` and `setFingerprint` methods in the constructor.

```
module.exports = class WorkerError extends BaseError {
  /**
   * Creates a new worker error.
   * @param {String} message Message for the error.
   * @param {Object} data Additional data for the error to be given to rollbar.
   * @param {Object} reporting Reporting options.
   * @param {String} queue Name of the queue for the worker.
   * @param {Object} job The job that was being processed.
   */
  constructor (message, data, reporting, queue, job) {
```